### DISCUSSION TITLE
Unique ID per application install

### DISCUSSION THREAD

**Discussion by NikolaGrgic** [tauri-apps/tauri#2818] [Q&A] [answered]
Hi all,

Is there a way to bind a unique ID to an install and then call upon it from the frontend, either via a default environment variable, JavaScript API, or the like? The way I'm imaging it is the following:

1. The user installs the app.
2. A random ID is generated and somehow securely stored in the app.
3. The ID can be called within the frontend.

Is a behavior like this possible with Tauri, or is there something already like that?

**Comment by nothingismagick** [comment, marked as answer]
There are many ways to do this, I would recommend integrating stronghold via our official plugin. It's generally not a wise move to use environment variables, at least if you are concerned about your user's security.

https://github.com/tauri-apps/tauri-plugin-stronghold

**Reply by NikolaGrgic** [reply]
You guys are great. Thanks a lot!
